Okay, nothing is ever a one line diff it seems .... the Windows failure is real (only with numpy 1.23.5):
```
[184/1446] Compiling C object scipy/linalg/_flapack.cp310-win_amd64.pyd.p/meson-generated_..__flapackmodule.c.obj
FAILED: scipy/linalg/_flapack.cp310-win_amd64.pyd.p/meson-generated_..__flapackmodule.c.obj 
"cc" "-Iscipy\linalg\_flapack.cp310-win_amd64.pyd.p" "-Iscipy\linalg" "-I..\scipy\linalg" "-IC:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\core\include" "-IC:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\f2py\src" "-IC:/hostedtoolcache/windows/Python/3.10.11/x64/Lib/site-packages/scipy_openblas32/include" "-IC:\hostedtoolcache\windows\Python\3.10.11\x64\Include" "-fvisibility=hidden" "-fdiagnostics-color=always" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-std=c17" "-O2" "-g" "-Wno-unused-but-set-variable" "-Wno-unused-function" "-Wno-conversion" "-Wno-misleading-indentation" "-mlong-double-64" "-D__USE_MINGW_ANSI_STDIO=1" "-DMS_WIN64=" "-Wno-empty-body" -MD -MQ scipy/linalg/_flapack.cp310-win_amd64.pyd.p/meson-generated_..__flapackmodule.c.obj -MF "scipy\linalg\_flapack.cp310-win_amd64.pyd.p\meson-generated_..__flapackmodule.c.obj.d" -o scipy/linalg/_flapack.cp310-win_amd64.pyd.p/meson-generated_..__flapackmodule.c.obj "-c" scipy/linalg/_flapackmodule.c
In file included from C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\core\include/numpy/ndarraytypes.h:1948,
                 from C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\core\include/numpy/ndarrayobject.h:12,
                 from C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\core\include/numpy/arrayobject.h:5,
                 from C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\f2py\src/fortranobject.h:13,
                 from scipy/linalg/_flapackmodule.c:23:
C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\core\include/numpy/npy_1_7_deprecated_api.h:14:9: note: '#pragma message: C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\numpy\core\include/numpy/npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION'
   14 | #pragma message(_WARN___LOC__"Using deprecated NumPy API, disable it with " \
      |         ^~~~~~~
scipy/linalg/_flapackmodule.c:101:10: fatal error: threads.h: No such file or directory
  101 | #include <threads.h>
      |          ^~~~~~~~~~~
compilation terminated.
```

--- a/meson.build
+++ b/meson.build
@@ -10,7 +10,7 @@ project(
   default_options: [
     'buildtype=debugoptimized',
     'b_ndebug=if-release',
-    'c_std=c99',
+    'c_std=c17',
     'cpp_std=c++17',
     'fortran_std=legacy',
     'blas=openblas',

--- a/scipy/meson.build
+++ b/scipy/meson.build
@@ -134,6 +134,15 @@ fortranobject_dep = declare_dependency(
   link_with: fortranobject_lib,
   include_directories: [inc_np, inc_f2py],
 )
+# Workaround for numpy#24761 on numpy<1.26.1 (see also gh-20515)
+_f2py_c_args = []
+if is_mingw
+  _f2py_c_args = '-DNPY_OS_MINGW'
+endif
+fortranobject_dep = declare_dependency(
+  dependencies: fortranobject_dep,
+  compile_args: _f2py_c_args,
+)
 
 # TODO: 64-bit BLAS and LAPACK
 #

